addons/website/models/website_page.py

```diff
@@ -340,6 +341,13 @@ def _search_fetch(self, search_detail, search, limit, order):
                 )
 
         def filter_page(search, page, all_pages):
+            # Exclude pages that do not pass ACL.
```

Comment:
Also, in their original report, the customer was talking about menu items that still show up too, nothing was to be done about that? :pray:

Reply:
I tried in 15.0 and 17.0, the menu filtering seems to work. On the screenshot, they are logged in - maybe their user did match the requirements to display the menu.

Tried scenario:
- "Signed in" only: hidden for visitor, shown for portal
- "Restricted group" + actual user: hidden for visitor and portal
- "Password": shown for visitor and portal (which seems reasonable - otherwise you cannot click on it to be requested the password)

I however removed the password ones from the search results.